### PR TITLE
Changes to Runtime Upgrades

### PR DESCRIPTION
**Pull request by joepetrowski**

### CODE REVIEW

File: docs/learn/learn-runtime-upgrades.md [hunk 1]
```diff
@@ -12,33 +12,13 @@ Runtime upgrades allow Polkadot to change the logic of the chain, without the ne
 
 ## Forking vs Forkless
```
Comment (profradha) [inline docs/learn/learn-runtime-upgrades.md]: Should the title still be Forking vs Forkless?

File: docs/learn/learn-runtime-upgrades.md [hunk 2]
```diff
@@ -56,56 +36,48 @@ Although upgrading your nodes is generally not necessary to follow an upgrade, w
...
 check the release notes, starting with the upgrade priority and acting accordingly.
 
-For general infrastructure providers, aside from following the Polkadot releases and upgrading in a
-timely manner, somethings to keep an eye out on is updating your parsing logic and acknowledging
-changes to the available RPC clients, such as the
+General infrastructure providers, aside from following the Polkadot releases and upgrading in a
+timely manner, should monitor changes to runtime events and auxilliary tooling, such as the
 [Substrate API Sidecar](https://github.com/paritytech/substrate-api-sidecar).
 
+Transactions constructed for runtime `n` will not work for runtimes `>n`. If a runtime upgrade occurs before broadcasting a priorly constructed transaction, you will need to reconstruct with the appropriate runtime version and corresponding metadata.
```
Comment (profradha) [inline docs/learn/learn-runtime-upgrades.md]: you will need to reconstruct **it** with...

File: docs/learn/learn-runtime-upgrades.md (hunk 1 above)
Comment (profradha) [reply docs/learn/learn-runtime-upgrades.md]: or just be **Forkless Upgrades**

**Review by DrW3RK — APPROVED** [review]
A few minor suggestions. The content as is should also be okay

File: docs/learn/learn-runtime-upgrades.md [hunk 3]
```diff
@@ -10,35 +10,24 @@ slug: ../learn-runtime-upgrades
 
 Runtime upgrades allow Polkadot to change the logic of the chain, without the need for a hard fork.
```
Comment (DannyS03) [inline docs/learn/learn-runtime-upgrades.md:11]: "logic of the chain" -> "chain's logic"

File: docs/learn/learn-runtime-upgrades.md [hunk 4]
```diff
@@ -10,35 +10,24 @@ slug: ../learn-runtime-upgrades
...
 
-## Forking vs Forkless
+## Forkless Upgrades
 
-When we talk about "forking", we are referring to the act of invoking the `fork` system call [within
-a system], where a process creates a copy of itself. In software development, this is the act of
-forking source code from one software package that creates a separate piece of that software.
+You may have come across the term "hard fork" before in the blockchain space. A **hard fork** occurs
+when a blockchain's logic changes such that nodes that do not include the new changes will not be
+able to remain in consensus with nodes that do. Such changes are backwards incompatible. Hard forks
```
Comment (salmad3) [inline docs/learn/learn-runtime-upgrades.md]: "backwards incompatible" ->"backward incompatible"

File: docs/learn/learn-runtime-upgrades.md [hunk 5]
```diff
@@ -10,35 +10,24 @@ slug: ../learn-runtime-upgrades
...
+can be political due to the nature of the upgrades, as well as logistically onerous due to the
+number (potentially thousands) of nodes in the network that need to upgrade their software.
 
-You may have come across the terms "hard fork" and "soft fork" before in the blockchain space.
+Rather than encode the runtime (a chain's "business logic") in the nodes, Polkadot nodes contain a
+WebAssembly [execution host](learn-polkadot-host). They maintain consensus on a very low level and
```
Comment (DannyS03) [inline docs/learn/learn-runtime-upgrades.md:22]: "very low level" -> "deficient"

File: docs/learn/learn-runtime-upgrades.md [hunk 6]
```diff
@@ -10,35 +10,24 @@ slug: ../learn-runtime-upgrades
...
+well established instruction set. The Polkadot runtime is stored on the Polkadot blockchain itself.
```
Comment (salmad3) [inline docs/learn/learn-runtime-upgrades.md]: "well established instruction set" -> "a well-established instruction set"

File: docs/learn/learn-runtime-upgrades.md [hunk 7]
```diff
@@ -10,35 +10,24 @@ slug: ../learn-runtime-upgrades
...
 
-A **hard fork** sees the blockchain's underlying technology (i.e. the protocol) radically changed
-and exists as a _different_ or _new_ version. This version is meant to be incompatible with the
-original version. This means that both blockchains exist simultaneously. Henceforth, a hard fork
-requires node operators to manually upgrade their nodes to the latest runtime version. In a
-distributed system, this is a complex process to coordinate and communicate.
+As such, Polkadot can upgrade its runtime by upgrading the logic stored on-chain, and removes the
```
Comment (salmad3) [inline docs/learn/learn-runtime-upgrades.md:25]: "upgrading the logic stored on-chain, and removes the" -> "upgrading the logic stored on-chain and remove the"

File: docs/learn/learn-runtime-upgrades.md [hunk 8]
```diff
@@ -56,56 +45,55 @@ Although upgrading your nodes is generally not necessary to follow an upgrade, w
...
 - API services
 - Node-as-a-Service (NaaS)
-- General infrastructure management (e.g. block explorers, exchanges)
+- General infrastructure management (e.g. block explorers, custodians)
+- [Wallets](../build/build-wallets.md)
 
 For validators, keeping in sync with the network is key. At times, upgrades will require validators
-to upgrade within a specific time frame to ensure continued sync with their node. It is essential to
-check the release notes, starting with the upgrade priority and acting accordingly.
+to upgrade their clients within a specific time frame, for example if a release includes breaking
```
Comment (salmad3) [inline docs/learn/learn-runtime-upgrades.md:61]: "for example" -> "for example,"

File: docs/learn/learn-runtime-upgrades.md [hunk 9]
```diff
@@ -56,56 +45,55 @@ Although upgrading your nodes is generally not necessary to follow an upgrade, w
...
+changes to networking. It is essential to check the release notes, starting with the upgrade
+priority and acting accordingly.
 
-For general infrastructure providers, aside from following the Polkadot releases and upgrading in a
-timely manner, somethings to keep an eye out on is updating your parsing logic and acknowledging
-changes to the available RPC clients, such as the
+General infrastructure providers, aside from following the Polkadot releases and upgrading in a
+timely manner, should monitor changes to runtime events and auxilliary tooling, such as the
```
Comment (salmad3) [inline docs/learn/learn-runtime-upgrades.md:66]: "upgrading in a timely manner, should monitor" -> "upgrading in a timely manner should monitor"

File: docs/learn/learn-runtime-upgrades.md [hunk 10]
```diff
@@ -56,56 +45,55 @@ Although upgrading your nodes is generally not necessary to follow an upgrade, w
...
 [Substrate API Sidecar](https://github.com/paritytech/substrate-api-sidecar).
 
+Transactions constructed for runtime `n` will not work for runtimes `>n`. If a runtime upgrade
+occurs before broadcasting a priorly constructed transaction, you will need to reconstruct it with
+the appropriate runtime version and corresponding metadata.
+
 ### For [Nominators](../maintain/maintain-guides-how-to-nominate-polkadot.md)
 
 Runtime upgrades don't require any actions by a nominator, though, it is always encouraged to keep
```
Comment (salmad3) [inline docs/learn/learn-runtime-upgrades.md]: "though, it is always encouraged to keep" -> "though it is always encouraged to keep"

File: docs/learn/learn-runtime-upgrades.md [hunk 11]
```diff
@@ -56,56 +45,55 @@ Although upgrading your nodes is generally not necessary to follow an upgrade, w
...
 up-to-date and participate with the latest runtime upgrade motions and releases, while keeping an
```
Comment (salmad3) [inline docs/learn/learn-runtime-upgrades.md:76]: "motions and releases, while keeping an" -> "motions and releases while keeping an"

File: docs/learn/learn-runtime-upgrades.md [hunk 12]
```diff
@@ -56,56 +45,55 @@ Although upgrading your nodes is generally not necessary to follow an upgrade, w
...
-
-1. `democracy(Started)` events and log `index` and `blockNumber`. Get
-   `pallets/democracy/storage/ReferendumInfoOf?key1=index&at=blockNumber` from `Sidecar` to get the
-   referendum info. It should have a status of `Ongoing`. Find the ending block number (`end`) and
-   the enactment `delay` (delay), where the execution block number will be `end + delay`.
+You can monitor the chain for upcoming upgrades. The client release notes include the hashes of any
+proposals related to any on-chain upgrades for easy matching. Monitor the chain for:
 
+1. `democracy(Started)` events and log `index` and `blockNumber`. This indicates that a referendum
+   has started (although does not mean that it is a runtime upgrade). Get the referendum info\*; it
```
Comment (salmad3) [inline docs/learn/learn-runtime-upgrades.md]: "(although does not mean that it is a runtime upgrade)" -> "(although, this does not mean that it is a runtime upgrade)"

File: docs/learn/learn-runtime-upgrades.md [hunk 13]
```diff
@@ -10,35 +10,24 @@ slug: ../learn-runtime-upgrades
...
-Polkadot sees itself as a next-generation blockchain because it addresses these obstacles in a
-meaningful way - revolutionizing the runtime upgrading process by enabling blockchains to upgrade
-themselves without the need to fork the chain. These forkless upgrades are enacted through
-Polkadot’s transparent [on-chain governance](learn-governance.md) system.
-
-## New [Client Release](https://github.com/paritytech/polkadot/releases)
+## New [Client Releases](https://github.com/paritytech/polkadot/releases)
 
 The existing runtime logic is followed to update the [Wasm](learn-wasm.md) runtime stored on the
 blockchain to a new version. The upgrade is then included in the blockchain itself, meaning that all
```
Comment (salmad3) [inline docs/learn/learn-runtime-upgrades.md:33]: In this section, the following: 
```
Nodes only need to be updated when the runtime requires new host functions or there is a change in
networking or consensus.
```
can be revised to: 
```
Nodes only require an update when the runtime requires new host functions or a change in
networking or consensus.
```

**Review by salmad3 — CHANGES_REQUESTED** [review]
Thanks @joepetrowski, content LGTM.
Left some writing suggestions. 

Also, we had added a pre-commit pretty check ([pretty-quick](https://prettier.io/docs/en/precommit.html) which runs pretty on changed/staged files. This allows for an easier way to ensure formatting and also creates an easy approach for external users (as it should run automatically). 
https://github.com/w3f/polkadot-wiki#style-guide

**Comment by salmad3** [comment]
When review is complete, merge is pending a build fix.

File: docs/learn/learn-runtime-upgrades.md (hunk 5 above)
Comment (joepetrowski) [reply docs/learn/learn-runtime-upgrades.md:22]: Why? Those don't mean the same thing.

File: docs/learn/learn-runtime-upgrades.md (hunk 4 above)
Comment (joepetrowski) [reply docs/learn/learn-runtime-upgrades.md]: Both are equally acceptable according to modern style guides.

File: docs/learn/learn-runtime-upgrades.md (hunk 9 above)
Comment (joepetrowski) [reply docs/learn/learn-runtime-upgrades.md:66]: No, it's the separation of a nonrestrictive clause.

File: docs/learn/learn-runtime-upgrades.md (hunk 7 above)
Comment (joepetrowski) [reply docs/learn/learn-runtime-upgrades.md:25]: No, it's the conjunction of two independent clauses. "comma and" is the correct way to handle it.

File: docs/learn/learn-runtime-upgrades.md (hunk 3 above)
Comment (joepetrowski) [reply docs/learn/learn-runtime-upgrades.md:11]: Why?

File: docs/learn/learn-runtime-upgrades.md (hunk 11 above)
Comment (joepetrowski) [reply docs/learn/learn-runtime-upgrades.md:76]: No, new clause.

File: docs/learn/learn-runtime-upgrades.md (hunk 12 above)
Comment (joepetrowski) [reply docs/learn/learn-runtime-upgrades.md]: No, there's no a reason to introduce an undeclared "this".

**Comment by joepetrowski** [comment]
> Also, we had added a pre-commit pretty check (pretty-quick which runs pretty on changed/staged files. This allows for an easier way to ensure formatting and also creates an easy approach for external users (as it should run automatically).

It doesn't seem to wrap lines though? We should fix the config so that it formats things properly.

**Review by salmad3 — APPROVED** [review]

**Pull request merged**